pkg/kubelet/kubelet_network_linux.go

```diff
@@ -51,111 +52,122 @@ const (
-	iptClients = append(iptClients, utiliptables.New(exec, utiliptables.ProtocolIPv6))
-
-	// and if they are not
-	if ipv6Primary {
-		protocols[0], protocols[1] = protocols[1], protocols[0]
-		iptClients[0], iptClients[1] = iptClients[1], iptClients[0]
+	iptClients := []utiliptables.Interface{
+		utiliptables.New(exec, utiliptables.ProtocolIPv4),
+		utiliptables.New(exec, utiliptables.ProtocolIPv6),
 	}
 
 	for i := range iptClients {
 		iptClient := iptClients[i]
-		if kl.syncNetworkUtil(iptClient) {
-			klog.InfoS("Initialized protocol iptables rules.", "protocol", protocols[i])
+		if kl.syncIPTablesRules(iptClient) {
+			klog.InfoS("Initialized iptables rules.", "protocol", iptClient.Protocol())
 			go iptClient.Monitor(
 				utiliptables.Chain("KUBE-KUBELET-CANARY"),
 				[]utiliptables.Table{utiliptables.TableMangle, utiliptables.TableNAT, utiliptables.TableFilter},
-				func() { kl.syncNetworkUtil(iptClient) },
+				func() { kl.syncIPTablesRules(iptClient) },
 				1*time.Minute, wait.NeverStop,
 			)
 		} else {
-			klog.InfoS("Failed to initialize protocol iptables rules; some functionality may be missing.", "protocol", protocols[i])
+			klog.InfoS("Failed to initialize iptables rules; some functionality may be missing.", "protocol", iptClient.Protocol())
+		}
+	}
+}
+
+// syncIPTablesRules ensures the KUBE-IPTABLES-HINT chain exists, and the martian packet
+// protection rule is installed. If the IPTablesCleanup feature gate is disabled it will
+// also synchronize additional deprecated iptables rules.
+func (kl *Kubelet) syncIPTablesRules(iptClient utiliptables.Interface) bool {
+	// Create hint chain so other components can see whether we are using iptables-legacy
+	// or iptables-nft.
+	if _, err := iptClient.EnsureChain(utiliptables.TableMangle, KubeIPTablesHintChain); err != nil {
+		klog.ErrorS(err, "Failed to ensure that iptables hint chain exists")
+		return false
+	}
+
+	if !iptClient.IsIPv6() { // ipv6 doesn't have this issue
+		// Set up the KUBE-FIREWALL chain and drop non-local packets to localhost if
+		// they're not part of an existing forwarded connection. See #90259
+		if _, err := iptClient.EnsureChain(utiliptables.TableFilter, KubeFirewallChain); err != nil {
+			klog.ErrorS(err, "Failed to ensure that filter table KUBE-FIREWALL chain exists")
+			return false
+		}
+
+		if _, err := iptClient.EnsureRule(utiliptables.Prepend, utiliptables.TableFilter, utiliptables.ChainOutput, "-j", string(KubeFirewallChain)); err != nil {
+			klog.ErrorS(err, "Failed to ensure that OUTPUT chain jumps to KUBE-FIREWALL")
+			return false
+		}
+		if _, err := iptClient.EnsureRule(utiliptables.Prepend, utiliptables.TableFilter, utiliptables.ChainInput, "-j", string(KubeFirewallChain)); err != nil {
+			klog.ErrorS(err, "Failed to ensure that INPUT chain jumps to KUBE-FIREWALL")
+			return false
+		}
+
+		if _, err := iptClient.EnsureRule(utiliptables.Append, utiliptables.TableFilter, KubeFirewallChain,
```

Comment:
Could you toss some helpful comments here?  The KEp itself does a decent job laying out why this is not in the "deprecated" bucket.